src/web-server.js

```diff
@@ -204,7 +207,36 @@ app.patch("/import-map.json", (req, res) => {
           services: req.body.imports,
           scopes: req.body.scopes,
         })
-        .then((newImportMap) => {
+        .then(async (newImportMap) => {
+          const imports = { ...req.body.imports };
+          const importKeys = Object.keys(imports);
+          const now = new Date().toISOString();
+          const { organization, environment, version, type, squad_name } =
+            req.body.metaData || {};
+          if (req.body.metaData) {
+            for (const key of importKeys) {
+              const params = {
+                created_at: now,
+                deployed_at: type === "deploy" ? now : undefined,
+                environment: environment,
+                link: imports[key],
+                organization: organization,
+                rollback_at: type === "rollback" ? now : undefined,
+                service_name: key,
```

Comment:
well put it here maybe can cause race condition, i think it break the `lock` they was build

because you put it in `modifyImportMap.then` and `releaseLock` was being call inside `modifyImportMap`

so for EX we call `t1` -> releaseLock -> call update hasura -> pending (at the time the `t2` can return resutl before t1